### PR TITLE
add base64 support for markdown

### PR DESCRIPTION
**Pull request by dberenbaum**
To support markdown reports in notebooks without needing to source from external files

### CODE REVIEW

File: src/dvc_render/vega.py [hunk 1]
```diff
@@ -120,5 +121,8 @@ def generate_markdown(self, report_path=None) -> str:
             plt.savefig(output_file)
             plt.close()
 
-            return f"\n![{self.name}]({output_file.relative_to(report_folder)})"
```
Comment (daavoo) [inline src/dvc_render/vega.py:123]: I think this actually breaks the integration with CML like:

https://github.com/iterative/example-get-started-experiments/blob/main/.github/workflows/run-studio-experiment.yml#L61

Could we make the base64 optional, default to the previous behavior, and explicitly enable in the DVCLive side only for notebook?

Comment (dberenbaum) [reply src/dvc_render/vega.py:123]: Updated to make the file optional and to return the markdown string if no file is passed. PTAL.

Comment (dberenbaum) [reply src/dvc_render/vega.py:123]: Updating dvclive to use this option. Should work the same as before as long as a file is passed.

File: src/dvc_render/markdown.py [hunk 2]
```diff
@@ -63,6 +65,9 @@ def render_markdown(
     for renderer in renderers:
         document.with_element(renderer.generate_markdown(report_path=output_path))
 
-    output_path.write_text(document.embed(), encoding="utf8")
+    if output_file and output_path:
```
Comment (daavoo) [inline src/dvc_render/markdown.py:68]:
```suggestion
    if output_path:
```

**Pull request merged**